lib/src/prepare_cronet.dart

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// Contains the nessesary setup code only. Not meant to be exposed.
+
+import 'dart:io' show Directory, File, Process, ProcessResult, ProcessStartMode;
+
+// import 'package:path/path.dart';
+
+import 'find_resource.dart';
+
+// TODO: Change URL and Version
+
+const _release = '1.0.0';
+const _cronetBinaryUrl =
+    'https://github.com/unsuitable001/dart_cronet_sample/releases/download/$_release/';
+final _cBinExtMap = {
+  'linux64': '.tar.xz',
+  'windows64': '.tar.gz',
+  'androidarm64-v8a': '.tar.xz',
+};
+
+const _cronetVersion = '"86.0.4240.198"';
+
+/// Builds the [wrapper] shared library
+/// according to [build.sh] file
+void buildWrapper() {
+  final wrapperPath = wrapperSourcePath();
+
+  print('Building Wrapper...');
+  var result =
+      Process.runSync('$wrapperPath/build.sh', [wrapperPath, _cronetVersion]);
+  print(result.stdout);
+  print(result.stderr);
+  print('Copying wrapper to project root...');
+  result = Process.runSync('cp', ['$wrapperPath/wrapper.so', '.']);
+  print(result.stdout);
+  print(result.stderr);
+}
+
+/// Place downloaded binaries to proper location
+void placeBinaries(String platform, String fileName) {
+  print('Extracting Cronet for $platform');
+  ProcessResult res;
+  // Process.runSync('mkdir', ['-p', 'cronet_binaries']);
+  if (platform.startsWith('windows')) {
+    res = Process.runSync('tar', ['-xvf', fileName]);
+  } else {
+    Directory('cronet_binaries').createSync();
+
+    // Do we have tar extraction capability
+    // in dart's built-in libraries?
+    res = Process.runSync('tar', ['-xvf', fileName, '-C', 'cronet_binaries']);
+  }
+
+  if (res.exitCode != 0) {
+    throw Exception(
+        "Can't unzip. Check if the downloaded file isn't corrupted");
+  }
+  print('Done! Cleaning up...');
+
+  File(fileName).deleteSync();
+  print('Done! Cronet support for $platform is now available!');
+}
+
+/// Download [cronet] library
+/// from Github Releases
+Future<void> downloadCronetBinaries(String platform) async {

```

Comment:
Can we get a `buildCronetBinaries` as well? (Automating the steps you did locally to build the binary.)

That way we can just run `buildCronetBinaries` when we want to make the binaries for a new version.

Also, that would enable users to build instead of download if they so prefer.

Reply:
Let's make this build utility to build the wrapper for that custom version and [print the link to build cronet](https://chromium.googlesource.com/chromium/src/+/refs/heads/main/components/cronet/build_instructions.md) in the console. Should we also take the responsibility for building actual cronet binaries? I guess that may turn out to be a little bit hard to maintain. What's your view?

Reply:
Well, you need to build the libcronet.xxxx.so/dll yourself also right? It would be useful to have the steps to do that as a Dart script, so that you don't forget.

We can either have that script in tool/update_cronet.dart so that you can update the shared libraries with a simple step, or we can have it available in bin/setup.dart as `dart run cronet:build-cronet` (or something similar).